src/main/java/seedu/address/logic/commands/CommentCommand.java

```diff
@@ -17,7 +17,7 @@
  * Adds a comment to an existing person in the address book.
  * AddCommentCommand is adapted from https://nus-cs2103-ay2122s2.github.io/tp/tutorials/AddRemark.html
  */
-public class CommentCommand extends Command {
+public class CommentCommand extends RedoableCommand {
```

Comment:
Why still extend `RedoableCommand` if it's doing the exact same thing as before?

Reply:
The extend `redoableCommand` enables the command to be able to undo and redo. If it would to be just extend 'Command' it does not allow command to be able to undo and redo

Reply:
if every command extends to `Command` then how do I only only allow undo and redo to specific commands? I believe not all command should be able to undo therefore i created a class to extend `Command` and added a methods to handle the undo and redo.

Reply:
Those who wants the command to be able to undo/redo, they can extend to `RedoableCommand` instead of `Command` which does not have the undo/redo

Reply:
The `redoableCommand` is only used to recognize that this command can be redo and undo in the logicManager to simplify the workload.

![image](https://user-images.githubusercontent.com/77482097/160979859-7aab58cc-cdfb-4107-87e8-61a93da49b53.png)

Reply:
I see.

However, now I am confused by why you need to do `saveSuccessMessage` and what benefits it brings? It doesn't seem like you're using the saved message either.

Also, the current implementation is incompatible with commands that open up GUI like `add`, `edit`, because you are saving to `undoRedoStack` whenever these commands are executed, instead of when data in `AddressBook` is changed. maybe you can look into some way around this?

Reply:
I think I am fine with merging this now since deadline is tomorrow. 
- I think `saveSuccessMessage` is never used, at least according to Intellij. 
- I cannot replicate your GUI bug. I think that it works with GUI add, edit on my side.
- However, there is a problem with the archive feature. It is buggy, but can fix in V1.4.

Reply:
> * I cannot replicate your GUI bug. I think that it works with GUI add, edit on my side.

Ah yes to replicate, here are the steps:
1. Type `add`
2. Close `AddWindow`
3. Type `undo`

No commands should be undone, and if any action should be taken, it would be to reopen `AddWindow` (but I think this one would be complicated)

Alternatively, if you were to add a new `Person`..
1. Type `add`
2. Fill in details of new `Person`
3. Submit
4. Type undo. It gives the correct behaviour of undoing the adding of a `Person`
5. Type undo again. It gives the wrong behaviour and reports `Undo Success!` when correct one should report `No more commands to undo!`, since typing `add` is just a intermediary to open up `AddWindow` rather than a command that modifies.